pandas/tests/io/json/test_pandas.py

```diff
@@ -792,7 +792,7 @@ def test_frame_from_json_precise_float(self):
 
     def test_typ(self):
         s = Series(range(6), index=["a", "b", "c", "d", "e", "f"], dtype="int64")
-        result = read_json(StringIO(s.to_json()), typ=None)
+        result = read_json(StringIO(s.to_json()), typ="series")
```

Comment:
Why did the tests need to change?

Reply:
This test didn't align with the docs/typing of `typ` accepting `None`. I added validation in this PR to ensure this case fails now

Reply:
Hmm OK. That's super strange we allowed that to begin with...but hopefully harmless taking it away